### ISSUE TITLE
[Bug] t2screen() can only put timestep that exists in the initial chart data.

### ISSUE THREAD

**Issue by arunavo4** [tvjsx/trading-vue-js#124]
## Description
When I am using t2screen() in an overlay to get x coordinates it works perfectly if the timestamp is one from the initial chart data.
it does not work if it's not from the initial chart data.

## Screenshot (if applicable)
![Screenshot from 2020-07-22 11-12-13](https://user-images.githubusercontent.com/28377631/88139038-89b16680-cc0c-11ea-883f-5938bdf4e7f5.png)

## Data Sample (if applicable)

The data sample must follow the current [data structure](https://github.com/C451/trading-vue-js/tree/master/docs/api#data-structure-new).

## Console errors (if any)
N/A
## Additional Info 

(if needed to reproduce the issue) 

(tvjs version, OS, browser, npm version...)
latest version 0.5.2

**Comment by arunavo4** [comment]
This happens when I try to get midpoint of two timesteps `let xm = layout.t2screen((p[1] + p[3])*0.5)` resulting in a timestep not in the initial chart data.

**Comment by C451** [comment]
Wow, I think I was done with this. Okay

**Comment by C451** [comment]
The chart looks quite messy overall. Just wondering, what's happened to the legend

**Comment by arunavo4** [comment]
These are custom overlays I made but it seems the legend is centred and not left-aligned. The RangeTrades Overlay is for drawing up the Trade ranges which I derived from the RangeTool.

**Comment by C451** [comment]
Interesting, Looks like I see this picture (with the broken legend) for the second time. If you could make a simple example to reproduce this, it would be great.

**Comment by arunavo4** [comment]
I tried using your playground thing. I really like it, but it wouldn't let me publish, and without publishing idk how to save the state and share

**Comment by arunavo4** [comment]
@C451 https://tvjs.io/play/?a=hfy0jtyg

**Comment by arunavo4** [comment]
How about `layout.t_magnet(t) // time -> nearest candle x` ? @C451

**Comment by arunavo4** [comment]
Looks like its more buggy

**Comment by C451** [comment]
@arunavo4 Nope, won't help, there is a bug in the algo https://github.com/tvjsx/trading-vue-js/blob/master/src/components/js/ti_mapping.js#L96 

This one is matching candles & overlay points step by step

**Comment by C451** [comment]
https://discord.com/channels/692731556288004116/692742783940689963/738351641924730960

The bug in action

**Comment by C451** [comment]
Looks like fixed in https://github.com/tvjsx/trading-vue-js/commit/44d7d1367303c83a2d006972f13b9312a6999322
To center the arrows perfectly use px coordinates.

**Comment by arunavo4** [comment]
@C451 This solves the problem, but yes they are not centred perfectly. Below is an example screenshot.

![Screenshot from 2020-08-20 17-03-58](https://user-images.githubusercontent.com/28377631/90765767-20566d80-e308-11ea-88e6-2aa7e030db85.png)

**Comment by C451** [comment]
@arunavo4 Yep, as they should not be. How to fix? Consider that an IQ test ; )
